reasoner/resolution/framework/Resolver.java

```diff
@@ -30,107 +29,59 @@
 public abstract class Resolver<T extends Resolver<T>> extends Actor.State<T> {
     private static final Logger LOG = LoggerFactory.getLogger(Resolver.class);
 
-    protected final String name;
-    private final Map<Request, ResponseProducer> responseProducers;
+    private final String name;
     private final Map<Request, Request> requestRouter;
-    private boolean isInitialised;
+    protected final ResolverRegistry registry;
 
-    public Resolver(Actor<T> self, String name) {
+    protected Resolver(Actor<T> self, String name, ResolverRegistry registry) {
```

Comment:
resolver registry is now passed in constructor, but should not be used until `initialiseDownstreamActors` is called